GCRCatalogs/alphaq.py

```diff
@@ -110,21 +110,30 @@ def _collect_native_quantities(name, obj):
             'sersic_disk':              'morphology/diskSersicIndex',
             'sersic_bulge':             'morphology/spheroidSersicIndex',
             'ellipticity_true':         'morphology/totalEllipticity',
-            'ellipticity_disk_true':    'morphology/diskEllipticity',
-            'ellipticity_bulge_true':   'morphology/spheroidEllipticity',
             'ellipticity_1_true':       'morphology/totalEllipticity1',
             'ellipticity_2_true':       'morphology/totalEllipticity2',
-            'ellipticity_1_disk_true':  'morphology/diskEllipticity1',
-            'ellipticity_2_disk_true':  'morphology/diskEllipticity2',
-            'ellipticity_1_bulge_true': 'morphology/spheroidEllipticity1',
-            'ellipticity_2_bulge_true': 'morphology/spheroidEllipticity2',
             'size_true': (
                 lambda size1, size2, lum1, lum2: ((size1*lum1)+(size2*lum2))/(lum1+lum2),
                 'morphology/diskMajorAxisArcsec',
-                'morphology/spheroidMinorAxisArcsec',
+                'morphology/spheroidMajorAxisArcsec',
                 'LSST_filters/diskLuminositiesStellar:LSST_r:rest',
                 'LSST_filters/spheroidLuminositiesStellar:LSST_r:rest',
             ),
+            'Av' : (lambda lum_v, lum_v_dust: (-2.5*(np.log10(lum_v_dust/lum_v))),
+                    'otherLuminosities/totalLuminositiesStellar:V:rest',
+                    'otherLuminosities/totalLuminositiesStellar:V:rest:dustAtlas',
+              
+            ),
+            'Rv' : (lambda lum_v, lum_v_dust, lum_b, lum_b_dust :
+                    np.choose(((lum_v_dust/lum_v==1.0) & (lum_b_dust/lum_b/lum_v_dust*lum_v==1.0)).astype(int),
```

Comment:
I don't think this code does what it should be? I could be wrong but in any case it's not easy to read.. I suggest create a function for it, like

```python
def _calc_Rv(lum_v, lum_v_dust, lum_b, lum_b_dust):
    v = lum_v_dust/lum_v
    b = lum_b_dust/lum_b
    Rv = np.log10(v) / np.log10(b/v)
    Rv[np.isnan(Rv)] = 1.0
    return Rv
```

Reply:
I was trying to keep it as a pure lambda function. I wasn't sure if messy lambda function was better than a free floating function. If you think a function is better, lets do that. 

I don't want to relying on NANs to handle the case of v=1 & b=1. A NAN can sneak it's way in some other way and we would just overwrite it instead of realizing something is off. The reason why I had (lum_b_dust/lum_b/lum_v_dust*lum_v ==1) is because there were rounding errors in the evaluation where v = 1, b != 1 but b/v = 1.

Reply:
@dkorytov thanks for the comment. Let's separate out a few things here ---

- lambda function vs. regular function: I don't really mind which one we use, and lambda function is indeed convenient for this kind of use. In this particular case, since there are two intermediate quantities (`v` and `b`) that are used more than once, I thought it would be more efficient to use a regular function in which we can define intermediate variables, but not in lambda function. 

- checking nan vs.  checking `(v == 1) & (b == 1)`: I agree with you point and we can use the latter. 

- the other main issue here is actually the use of `np.choose`, I think you might have meant `np.where`? In any case, using a boolean mask to slice and then set values is probably the most efficient and readable way. 

So, one way to implement would be: 
```python
def _calc_Rv(lum_v, lum_v_dust, lum_b, lum_b_dust):
    v = lum_v_dust/lum_v
    b = lum_b_dust/lum_b
    b /= v
    Rv = np.log10(v) / np.log10(b)
    Rv[(v == 1) & (b == 1)] = 1.0
    return Rv
```

Reply:
I agree with all points. I forgot about `np.where`.  `np.choose` is a messier solution.

Reply:
There are still NANs when b = 0 and v = 0, which happens when all the v/b band are completely quenched by dust. Should we add another condition to avoid this known NAN? Or just let the NANs be until we figure out what to do with v = b = 0 issue.

Reply:
In the case that `lum_v_dust/lum_v = lum_b_dust/lum_b = 0`, shouldn't we set `A_v = np.inf` and `R_v = 1`?  

But @danielsf should comment on this.

Reply:
It depends at the rate `lum_v_dust/lum_v` and `lum_b_dust/lum_b` go to zero. If they have different rates, we can get Rv = inf/real number = +/- inf. If it's the same rate, then it should equal 1.

Reply:
I think I would prefer setting `Av=Rv=NaN` in these cases.  That way, on the CatSim side, we can just detect these objects with `np.where(np.isnan())`, as opposed to `np.where(np.logical_or(np.isnan(), np.isinf())))`.

Do I infer correctly that the long term goal is to prevent `lum_v_dust` and `lum_b_dust` from ever actually going to zero (or, at least, minimizing the frequency with which this happens)?  Or do we think this corresponds to some actual physical effect?